src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,246 @@
+)
+@click.option("--assignee", required=False, help="Filter by user UID of assignee.")
+@click.option(
+    "--min-create-time",
+    required=False,
+    help="Fetch cases created after given date time.",
+)
+@click.option(
+    "--max-create-time",
+    required=False,
+    help="Fetch cases created before given date time.",
+)
+@click.option(
+    "--min-update-time",
+    required=False,
+    help="Fetch cases last updated after given date time.",
+)
+@click.option(
+    "--max-update-time",
+    required=False,
+    help="Fetch cases last updated before given date time.",
+)
+@click.option("--status", required=False, help="Filter cases by case status.")
+@format_option
+@sdk_options()
+def _list(
+    state,
+    name,
+    assignee,
+    subject,
+    min_create_time,
+    max_create_time,
+    min_update_time,
+    max_update_time,
+    status,
+    format,
+):
+    """List all the cases."""
+    pages = state.sdk.cases.get_all(
+        name=name,
+        assignee=assignee,
+        subject=subject,
+        min_create_time=min_create_time,
+        max_create_time=max_create_time,
+        min_update_time=min_update_time,
+        max_update_time=max_update_time,
+        status=status,
+    )
+    formatter = OutputFormatter(format, _HEADER_KEYS_MAP)
+    cases = _parse_cases(pages)
+    if cases:
+        formatter.echo_formatted_list(cases)
+
+
+@cases.command()
+@case_number_arg
+@sdk_options()
+def show(state, case_number):
+    """Show case details."""
+    click.echo(state.sdk.cases.get_case(case_number))
+
+
+@cases.command()
+@case_number_arg
+@click.option(
+    "--path", type=str, help="File path. Defaults to current directory", default="."
+)
+@sdk_options()
+def export(state, case_number, path):
+    """Download case detail summary in a pdf file at the given path with name <case_number>_case_summary.pdf."""
+    response = state.sdk.cases.export_summary(case_number)
+    with open("{}/{}_case_summary.pdf".format(path, case_number), "wb") as f:
+        f.write(response.content)
+
+
+@cases.group(cls=OrderedGroup)
+@sdk_options()
+def file_events(state):
+    """Fetch file events associated with the case."""
+    pass
+
+
+@file_events.command("list")
+@case_number_arg
+@sdk_options()
+def file_events_list(state, case_number):
+    """List all the events associated with the case."""
+    try:
+        response = state.sdk.cases.file_events.get_all(case_number)
+    except Py42NotFoundError as err:
+        click.echo("Invalid case-number.")
+        raise err
+    if not response["events"]:
+        click.echo("No events found.")
+    for event in response["events"]:
+        click.echo(event)
+
+
+@file_events.command("show")
+@case_number_arg
+@file_event_id_option
+@sdk_options()
+def _show(state, case_number, event_id):
+    """Show event details for the given event id associated with the case."""
+    try:
+        click.echo(state.sdk.cases.file_events.get_event(case_number, event_id))
```

Comment:
Show command are supposed to be more table like, and not just echoing JSON. See other show commands in the CLI.